### ISSUE TITLE
DuQ scale and offset parameter

### ISSUE THREAD

**Issue by MarsJacobs** [EunhyeokPark/PROFIT#5]
Hi! Thank you for sharing this great pytorch code. It helps me a lot.
I have a question about the Duq Quantization method and its code.

In your PROFIT paper, "_Our proposed DuQ method allows us to utilize the full value range of activation including the negative ones, and to achieve that, the discretized data can be mapped ro an arbitraty range throug a _**scale alpha' and offset beta**_..."_ 

![스크린샷 2021-02-09 오후 12 02 36](https://user-images.githubusercontent.com/54992207/107309917-b5277000-6ace-11eb-880e-e2505bca2a32.png)

What I understand is, in post transformation stage, quantization dynamic range can be moved to full range using alpha' and beta value.

But in duq.py code, I couldn't find the parameter which acts as offset beta.

In Q_HSwish Class, 
![스크린샷 2021-02-09 오전 11 55 12](https://user-images.githubusercontent.com/54992207/107309387-ae4c2d80-6acd-11eb-8cfd-df343895999b.png)

After multiply c value, which is same as alpha' in post transformation stage, I think self.d is same as beta. But self.d is constant, not the parameter like value a and c.

Maybe I couldn't understand the concept of DuQ quantization. What exactly beta do in post transformation stage and is the self.d same as beta?

Thank you.

**Comment by EunhyeokPark** [comment]
Hi, that's a very good question. According to my experiments, learnable alpha increases the accuracy a lot, but learnable beta only gives substantial improvement. Meanwhile, the fixed beta, i.e., self.d in the code, gives negligible accuracy drop, but the training becomes much robust to the hyper-parameter setting. I recommend you to use the fixed beta for the easy-to-use implementation, but if you want, you can easily modify the code with the learnable beta.
